### PR TITLE
BlobDB for storing attachments outside of couch

### PR DESCRIPTION
**Pull request by millerdev**
This implements a filesystem blob database and a mixin that adapts the couchdbkit attachments API to the new blob database. None of this code is being used yet, but it will be soon, hopefully.

In the future:
```py
from dimagi.ext.attachments import BlobMixin

class SavedBasicExport(BlobMixin, Document):
    _migrating_from_couch = True # temporary during migration phase

    # adapt methods to use new `blobs` property
    ...
```
Additional things to be implemented:
- A migration to transfer attachments out of couchdb into the new blob storage backend.
- A pillow to clean up blobs attached to deleted couch documents.

@dannyroberts @snopoke cc @kaapstorm

### CODE REVIEW

**Comment by dannyroberts** [comment]
This looks great!

Any idea on the test failure?

File: dimagi/ext/tests/test_attachments.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        doc = self.make_doc(FallbackToCouchDocument)
+        doc._attachments[name] = {"content": b"couch content"}
+        assert doc.delete_attachment(name), "couch attachment not deleted"
+        assert name not in doc._attachments, doc._attachments
+
+@generate_cases([
+    (None, None),
+    ("abc", None),
+    (None, "abc"),
+], TestBlobMixin)
```
Comment (snopoke) [inline dimagi/ext/tests/test_attachments.py]: :nose: :+1:

**Comment by kaapstorm** [comment]
Very cool.

**Comment by czue** [comment]
:+1: nice!

**Comment by millerdev** [comment]
:white_check_mark:

**Comment by millerdev** [comment]
Made one more commit I decided to throw in here. Unfortunately that means we need to wait on another test run.

**Comment by millerdev** [comment]
:+1:

**Comment by millerdev** [comment]
@dannyroberts helped me realize this can all be moved into the [hq](https://github.com/dimagi/commcare-hq/)

**Pull request closed without merging**